### PR TITLE
Maint: precommit clang-format

### LINKED ISSUE #935: pre-commit doesn't clang-format
pre-commit doesn't call clang-format, so it's still possible to make commits that fail the linter

### PR DESCRIPTION
**Pull request by hoodmane**
This resolves #935. I also changed the `lint` make recipe to use `find` which is more resilient to changes in directory structure and doesn't output those annoying `No such file or directory` messages to stdout. 

I changed the name of the `apply-lints` recipe to `apply-lint` for consistency, and I updated it to only apply lints to files with staged changes.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ SHELL := /bin/bash
 CC=emcc
 CXX=em++
 OPTFLAGS=-O2
-CFLAGS=$(OPTFLAGS) -g -I$(PYTHONINCLUDE) -Wno-warn-absolute-paths
+CFLAGS=$(OPTFLAGS) -g -I$(PYTHONINCLUDE) -Wno-warn-absolute-paths -D TEST
```
Comment (dalcde) [inline Makefile:17]: This change is probably accidentally leaked from #937?

Comment (hoodmane) [reply Makefile:17]: Yup, thanks.

File: Makefile [hunk 2]
```diff
@@ -162,7 +162,7 @@ clean-all: clean
 	rm -fr cpython/build
 
 %.bc: %.c $(CPYTHONLIB)
-	$(CC) -o $@ -c $< $(CFLAGS) -Isrc/type_conversion/
+	$(CC) -o $@ -c $< $(CFLAGS) $(EXTRA_CFLAGS) -Isrc/type_conversion/
```
Comment (dalcde) [inline Makefile:165]: This also seems leaked

**Comment by hoodmane** [comment]
@dalcde How does this look to you? Can this be merged?

**Comment by dalcde** [comment]
`clang-format-precommit.sh` complains when `$FILES` is empty. Also, would you like to move it into `tools/`?

**Comment by dalcde** [comment]
Well it prints an error message but still returns 0, so I guess good enough? The same happens with `apply-lints`. I'm not sure how I feel about `apply-lints` only applying to staged commits; my personal workflow applies lint before staging.

**Comment by hoodmane** [comment]
> I'm not sure how I feel about apply-lints only applying to staged commits; my personal workflow applies lint before staging.

That sounds reasonable, we can change it. I did not think very carefully about this stuff.

**Comment by hoodmane** [comment]
I made apply-lints to be compatible with the behavior of precommit which only cares about staged files.

**Comment by hoodmane** [comment]
@dalcde How do you like it now?

**Pull request merged**